test/assert.js

```diff
@@ -268,12 +268,46 @@ test('.same() should not mask RangeError from underlying assert', function (t) {
 	var b = new Circular();
 
 	t.throws(function () {
-		assert.notSame(a, b);
+		assert.notDeepEqual(a, b);
 	});
 
 	t.doesNotThrow(function () {
-		assert.same(a, b);
+		assert.deepEqual(a, b);
 	});
 
 	t.end();
 });
+
+test('.same() should log a deprecation notice', function (t) {
+	var calledWith;
+	var originalConsole = console.warn;
+	console.warn = function () {
```

Comment:
Looks like deprecate calls an undocumented method [`process.emitWarning`](https://github.com/nodejs/node/blob/master/lib/internal/util.js#L18-L24).

That ends up causing a [`warning` event to be emitted on `process`](https://github.com/nodejs/node/blob/master/lib/internal/process/warning.js#L47), which seems to be [handled by writing to `console.error`](https://github.com/nodejs/node/blob/master/lib/internal/process/warning.js#L18-L24).

So try `console.error`?

Or just trust that `util.deprecate` works.